### PR TITLE
Remove SchemaQualifiedName

### LINKED ISSUE #1072: Remove SchemaQualifiedName
In some parts of the code we use this and in other parts we use schema and table name separately. In some places it makes sense to treat them separately, such as in the code that accesses and manipulates annotations. However, when it comes to code that is reasoning about table names in relational code we should probably consistently use this class or if it doesn't make sense then get rid of it.

### PR DESCRIPTION
**Pull request by smitpatel**
Resolves #1072

### CODE REVIEW

**Comment by smitpatel** [comment]
Parsing method in SchemaQualifiedName was being used in unit testing only. Is that required? If yes where it should be put?

File: src/EntityFramework.Relational/SqlGenerator.cs [hunk 1]
```diff
@@ -71,23 +73,26 @@ public abstract class SqlGenerator
...
 
         public virtual void AppendInsertCommand(
             [NotNull] StringBuilder commandStringBuilder,
-            SchemaQualifiedName schemaQualifiedName,
+            [NotNull] string tableName,
+            [CanBeNull] string schemaName,
             [NotNull] IReadOnlyList<ColumnModification> writeOperations)
         {
             Check.NotNull(commandStringBuilder, "commandStringBuilder");
+            Check.NotNull(tableName, "tableName");
```
Comment (bricelam) [inline src/EntityFramework.Relational/SqlGenerator.cs]: We typically use `Check.NotEmpty()` for strings.

File: src/EntityFramework.Relational/Update/ModificationCommand.cs [hunk 2]
```diff
@@ -147,5 +151,15 @@ public virtual void PropagateResults([NotNull] IValueReader reader)
...
+
+        public virtual string EscapedTableName => Escape(TableName);
+        public virtual string EscapedSchemaName => Escape(SchemaName);
+
+        private static string Escape(string name)
+        {
+            return name.IndexOfAny(new[] { ']', '[', '.' }) != -1
+                ? "[" + name.Replace("]", "]]") + "]"
+                : name;
+        }
```
Comment (bricelam) [inline src/EntityFramework.Relational/Update/ModificationCommand.cs]: Are these used anywhere? If not, remove.

**Comment by bricelam** [comment]
No need to keep the parsing functionality.

**Comment by bricelam** [comment]
After addressing feedback, :shipit:

File: src/EntityFramework.Relational/Update/ModificationCommand.cs (hunk 2 above)
Comment (smitpatel) [inline src/EntityFramework.Relational/Update/ModificationCommand.cs]: There is a test which uses them. Though test works without escaping also. Do we need escaping code somewhere while generating sql?

Comment (bricelam) [inline src/EntityFramework.Relational/Update/ModificationCommand.cs]: We have escaping elsewhere. This type of escaping is SQL Server-specific anyway and doesn't belong in Relational.

**Pull request merged**